test/framework/machinedeployment_helpers.go

```diff
@@ -255,7 +259,10 @@ func WaitForMachineDeploymentRollingUpgradeToStart(ctx context.Context, input Wa
 	log.Logf("Waiting for MachineDeployment rolling upgrade to start")
 	Eventually(func() bool {
 		md := &clusterv1.MachineDeployment{}
-		Expect(input.Getter.Get(ctx, client.ObjectKey{Namespace: input.MachineDeployment.Namespace, Name: input.MachineDeployment.Name}, md)).To(Succeed())
+		err := input.Getter.Get(ctx, client.ObjectKey{Namespace: input.MachineDeployment.Namespace, Name: input.MachineDeployment.Name}, md)
```

Comment:
Please change this to the following instead:
```go
Eventually(func(g Gomega) bool {
...
g.Expect
```

This way the Expect does not fail the Eventually immediately (please also check other places)

Reply:
I'm not sure what you mean. The current change returns `false` in the `err != nil` condition, which will induce a sleep + retry assuming the timeout hasn't been reached.

Reply:
is there work to do here? (happy to make changes, apologies if I still don't understand the ask!)

Reply:
Sorry KubeCon, should have set my GitHub status accordingly.. :) 

I would suggest the following:
```go
Eventually(func(g Gomega) bool {
		md := &clusterv1.MachineDeployment{}
		g.Expect(input.Getter.Get(ctx, client.ObjectKey{Namespace: input.MachineDeployment.Namespace, Name: input.MachineDeployment.Name}, md)).To(Succeed())
		return md.Status.Replicas != md.Status.AvailableReplicas
	}, intervals...).Should(BeTrue())
```

By handing over Gomega and using it via `g.Expect` the `Expect` will only fail the test after the timeout from eventually is reached.

I think it would be a bit more elegant this way and would surface the error better when the `Eventually` eventually fails.

(If that suggestion is fine for you, please also adjust other cases on this PR)